### PR TITLE
Add egl rendering target

### PR DESCRIPTION
**Pull request by neithernut**
This PR introduces a rendering target for EGL buffers.
It's intended to replace the blitting code we have atm through it's blitting method.

### CODE REVIEW

**Comment by neithernut** [comment]
The last commit adds a MD file describing how I intend to handle buffers/surfaces. At the end, it roughly explains two options to do this, technically. Feel free to comment, especially on the two options suggested.

File: src/compositor/buffer/egl_target.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ * @extends ws_egl_buffer
+ */
+struct ws_egl_target {
+    struct ws_egl_buffer egl;   //!< @protected base class
+    struct ws_set surfaces;     //!< @protected shell surfaces blitted on here
+    struct ev_async watcher;    //!< @protected watcher to use for updates
+};
+
+/**
+ * Variable which holds the type information about the ws_egl_targte type
```
Comment (matthiasbeyer) [inline src/compositor/buffer/egl_target.h]: Typo.

**Comment by matthiasbeyer** [comment]
I think I'm fine with this, but I'd like you to wait for @TheNeikos comment.

**Comment by neithernut** [comment]
@TheNeikos reviewed it and said it was feasible, we will, however, perform some inqueries on the subject and give me more feedback. I _may_ postpone implementation of the (shared) texture until then.

**Comment by neithernut** [comment]
Mh... some header is missing, ...
Anyways, from this point on, I think I'll have to rely on advise from @TheNeikos  once again. (texture binding etc)
Furthermore, I'll probably split this PR into two.

**Comment by neithernut** [comment]
One question which has yet to be solved is whether binding a texture is thread safe or whether OpenGL needs to be locked globally.

**Comment by neithernut** [comment]
The entire texture area stuff was moved to another PR, which blocks this one.

**Comment by neithernut** [comment]
Turns out this is not needed.

**Pull request closed without merging**